autogpt_platform/autogpt_libs/autogpt_libs/supabase_integration_credentials_store/store.py

```diff
@@ -100,6 +105,29 @@ async def store_state_token(self, user_id: str, provider: str) -> str:
 
         return token
 
+    async def get_scopes_from_state_token(
+        self, user_id: str, token: str, provider: str
+    ) -> list[str]:
+        user_metadata = self._get_user_metadata(user_id)
+        oauth_states = user_metadata.get("integration_oauth_states", [])
+
+        now = datetime.now(timezone.utc)
+        valid_state = next(
+            (
+                state
+                for state in oauth_states
+                if state["token"] == token
+                and state["provider"] == provider
+                and state["expires_at"] > now.timestamp()
+            ),
+            None,
+        )
+
+        if valid_state:
+            return valid_state.get("scopes", [])
+
+        return []
```

Comment:
1. Any reason not to combine this with `verify_state_token`?
2. Returning `[]` regardless whether that's the actual list of scopes or the state is no longer valid or doesn't exist seems odd

Reply:
conceptually, the token would have no scopes if none of them are valid

Reasoning for it existing is you may want to check what scopes you have before trying to use the token